Redesigned sign in view controller

lexypaul13 opened the pull request:
## Removed goback button

## Changed the font on the user segmented control using FontFamily enum

## Changed color type on the view controller

![Simulator Screen Shot - iPhone 13 - 2021-11-01 at 11 29 00](https://user-images.githubusercontent.com/55071531/139697375-10a84acb-7c13-4271-a22b-b1ea5208972f.png)
.
<img width="534" alt="Screen Shot 2021-11-01 at 11 30 59 AM" src="https://user-images.githubusercontent.com/55071531/139697469-365c5b69-350f-4847-9c62-122a6fed99b0.png">

<img width="1036" alt="Screen Shot 2021-11-01 at 11 31 18 AM" src="https://user-images.githubusercontent.com/55071531/139697522-24fa99e5-b9dc-4c20-b49b-28c000c0a6db.png">

Commit:
Redesigned sign in view controller
Added two icons to the assets folders

Commit:
Changed file type of text field container to weak

Commit:
Removed go back button

Changed the font on the user segmented control using FontFamily enum

Used UIcolor for the view controller.

Changed file type for the textfield container to weak

AlexCornforth commented on TreeTracker/UI/Views/Authentication/SignIn/SignInViewController.swift:
```diff
@@ -23,24 +24,37 @@ class SignInViewController: UIViewController, KeyboardDismissing, AlertPresentin
-            usernameSegmentedControl.setImage(Asset.Assets.mail.image, forSegmentAt: 1)
-            usernameSegmentedControl.setTitleTextAttributes([.foregroundColor: Asset.Colors.grayDark.color], for: .selected)
-            usernameSegmentedControl.setTitleTextAttributes([.foregroundColor: Asset.Colors.grayLight.color], for: .normal)
+            let attr = NSDictionary(object: UIFont(name: "Montserrat-Bold", size: 16)!, forKey: NSAttributedString.Key.font as NSCopying)
```
You could use the swiftgen generated FontFamily here. That will also remove the force cast

lexypaul13 replied:
Hey Alex,  
Should it look like this ? 

let attr = NSDictionary(object: FontFamily.Montserrat.bold.font(size: 16) ?? "", forKey: NSAttributedString.Key.font as NSCopying)
             usernameSegmentedControl.setTitleTextAttributes(attr as? [ NSAttributedString.Key : AnyObject ], for: .normal)

AlexCornforth replied:
This shold do it:
```
let attributes: [NSAttributedString.Key: Any] = [.font: FontFamily.Montserrat.bold.font(size: 16.0)]
usernameSegmentedControl.setTitleTextAttributes(attributes, for: .normal)
```

AlexCornforth replied:
Not sure you need the extra casting

lexypaul13 replied:
Thanks

AlexCornforth commented on TreeTracker/UI/Views/Authentication/SignIn/SignInViewController.swift:
```diff
@@ -69,7 +84,7 @@ private extension SignInViewController {
     @IBAction func logInButtonPressed() {
         viewModel?.login()
-    }
+     }
```
Nitpick: Could remove the space added here

lexypaul13 replied:
I've cleared that

AlexCornforth commented on TreeTracker/UI/Views/Authentication/SignIn/SignInViewController.swift:
```diff
@@ -50,6 +64,7 @@ class SignInViewController: UIViewController, KeyboardDismissing, AlertPresentin
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        self.view.backgroundColor = UIColor.black
```
I think we should add background color to the Colors.xcassets (which we can set as black). That way we can have one place where we configure the background color. Happy to assist with this if you need.

AlexCornforth commented on TreeTracker/UI/Views/Authentication/SignIn/SignInViewController.swift:
```diff
@@ -23,24 +24,37 @@ class SignInViewController: UIViewController, KeyboardDismissing, AlertPresentin
         }
     }
-    @IBOutlet private var loginButton: PrimaryButton! {
+    @IBOutlet private var goBackButton: UIButton! {
```
Shall we remove this go back button for now as we are not going to need it.

lexypaul13 replied:
I've removed that.

AlexCornforth replied:
@lexypaul13 could you remove the back button code too?

AlexCornforth commented on TreeTracker/UI/Views/Authentication/SignIn/SignInViewController.swift:
```diff
@@ -23,24 +24,37 @@ class SignInViewController: UIViewController, KeyboardDismissing, AlertPresentin
+            usernameTextField.clipsToBounds = true
         }
     }
     @IBOutlet private var usernameSegmentedControl: UISegmentedControl! {
```
We should also set the text on the segmented control here too. It should be added tho the Localizable.strings file so swiftgen can generate the L10n enums which we can the use here.

lexypaul13 replied:
What part of the segmented controller should I set the text ?

![Simulator Screen Shot - iPhone 13 - 2021-11-01 at 11 29 00](https://user-images.githubusercontent.com/55071531/140403141-65f9ec0f-f1f1-4a58-a840-c38968a0a5f1.png)
?

AlexCornforth replied:
Forogt these had to text - ignore this

AlexCornforth commented on TreeTracker/UI/Views/Authentication/SignIn/SignInViewController.swift:
```diff
@@ -23,24 +24,37 @@ class SignInViewController: UIViewController, KeyboardDismissing, AlertPresentin
-            usernameSegmentedControl.setTitleTextAttributes([.foregroundColor: Asset.Colors.grayLight.color], for: .normal)
+            let attr = NSDictionary(object: UIFont(name: "Montserrat-Bold", size: 16)!, forKey: NSAttributedString.Key.font as NSCopying)
+            usernameSegmentedControl.setTitleTextAttributes(attr as? [ NSAttributedString.Key : AnyObject ], for: .normal)
+            if #available(iOS 13.0, *) {
```
How does this look in iOS11/12

lexypaul13 replied:
Hey Alex, are you referring to this ?If so, I couldn't find what the ios11/12 version of it looked like.
if #available(iOS 13.0, *) {
                usernameSegmentedControl.selectedSegmentTintColor = Asset.Colors.primaryGreen.color
            } 
I

AlexCornforth replied:
@lexypaul13 are you able to install older simulators and run on those?

lexypaul13 replied:
Yes, here's a screen shot of it on the iPhone 8
![Simulator Screen Shot - iPhone 8 - 2021-11-08 at 15 13 43](https://user-images.githubusercontent.com/55071531/140811359-5c7a96c7-836c-47a1-a436-c3c4f5a13b87.png)
.

lexypaul13 replied:
@AlexCornforth Is this color okay, and is it appropriate to change the color by the key path
<img width="1160" alt="Screen Shot 2021-11-15 at 10 27 30 AM" src="https://user-images.githubusercontent.com/55071531/141808868-fd96e3ad-7f42-40e2-bbac-9e4dcabdad79.png">
?

AlexCornforth replied:
I think it would be better to set the placeholder colour in code - i generally just use storyboards for layout and nothing else. More storyboard changes = increased risk of nasty conflicts. Also code is much easier to review than the giant storyboard xml files.

lexypaul13 replied:
Hey Alex, I've changed the placeholder text color using the code below.
UsernameTextField.attributedPlaceholder = NSAttributedString( string: L10n.SignIn.TextInput.PhoneNumber.placeholder,attributes: [NSAttributedString.Key.foregroundColor: Asset.Colors.grayLight.color]
           
![Simulator Screen Shot - iPhone 13 - 2021-11-29 at 14 27 17](https://user-images.githubusercontent.com/55071531/143930188-c90c5210-cdbf-4938-962c-704de538b448.png)
 )

AlexCornforth reviewed (changes requested):
Thanks for this @lexypaul13. I've left some comments to address. Let me know if want me to go through any of them in more detail.

AlexCornforth commented:
@lexypaul13 i think the placeholder text color needs to be lighter in the phone/email field. Its quite hard to see right now

Commit:
Removed back button.

Changed placedholder text color.

Used FontFamily for text on User Segmented control .

Commit:
Changed placeholder text color in code

AlexCornforth reviewed (approved):
Nice work @lexypaul13 thanks!
